src/leveled_penciller.erl

```diff
@@ -1613,338 +1615,236 @@ compare_to_sqn(ObjSQN, _SQN) when is_integer(ObjSQN) ->
 compare_to_sqn(Obj, SQN) ->
     compare_to_sqn(leveled_codec:strip_to_seqonly(Obj), SQN).
 
+-spec maybelog_fetch_timing(
+    leveled_monitor:monitor(),
+    memory|leveled_pmanifest:lsm_level(),
+    leveled_monitor:timing(),
+    boolean()) -> ok.
+maybelog_fetch_timing(_Monitor, _Level, no_timing, _NF) ->
+    ok;
+maybelog_fetch_timing({Pid, _StatsFreq}, _Level, FetchTime, true) ->
+    leveled_monitor:add_stat(Pid, {pcl_fetch_update, not_found, FetchTime});
+maybelog_fetch_timing({Pid, _StatsFreq}, Level, FetchTime, _NF) ->
+    leveled_monitor:add_stat(Pid, {pcl_fetch_update, Level, FetchTime}).
 
 %%%============================================================================
-%%% Iterator functions 
-%%% 
-%%% TODO - move to dedicated module with extended unit testing
+%%% Key folder
 %%%============================================================================
 
+-type sst_iterator()
+    :: #{
+        leveled_pmanifest:lsm_level() =>
+            list(leveled_sst:expandable_pointer()|leveled_codec:ledger_kv()),
+        -1 =>
```

Comment:
Not sure you really want to put an integer here. The nice thing of Erlang maps is that you could use an atom like: `filteredL0`(see line 824)

Reply:
but -1 is most likely also the thing you use below when you do integer comparison and then it makes sense.

Reply:
The other keys in the map are integers >=0 representing the level in the lsm tree - but this is the level above level 0 -> hence level -1